transport.go

```diff
@@ -398,6 +408,15 @@ func (t *Transport) handlePacket(p receivedPacket) {
 		t.handleNonQUICPacket(p)
 		return
 	}
+	if len(p.data) > int(t.MaxUDPPayloadSize) {
```

Comment:
This is inconsistent with how we handle packets larger 1452 bytes. How much effort would it be to wire this up properly, such that we pass a correctly sized buffer to the kernel? Would that create any problems with GRO?

Reply:
Are you asking if we can pipe this config directly into areas like [sys_conn.go#L94](https://github.com/chungthuang/quic-go/blob/b73a4de7eaa87773d9da570442752470b051ace6/sys_conn.go#L94), or [send_stream.go#L211](https://github.com/chungthuang/quic-go/blob/723a188dde0db4989e988749991fd424b8a0daab/send_stream.go#L211)?
I don't think any of those areas need to change as long as we make sure that the transport parameter `MaxUDPPayloadSize` provided is strictly less than the `MaxPacketBufferSize`? I understand that the RFC expects to allow the transport parameter to be set to a maximum of 65527, but I don't think that would work anyways regarding how the internal stream frame pools are statically allocated to block sizes of 1452.

Could you also point to a specific areas of where handling packets larger than the 1452? I could be missing it completely.

Regarding the GRO comment, similarly, if we make sure that the transport parameter is strictly less than the `MaxPacketBufferSize` of 1452, this shouldn't be a problem?

Reply:
> Are you asking if we can pipe this config directly into areas like [sys_conn.go#L94](https://github.com/chungthuang/quic-go/blob/b73a4de7eaa87773d9da570442752470b051ace6/sys_conn.go#L94), or [send_stream.go#L211](https://github.com/chungthuang/quic-go/blob/723a188dde0db4989e988749991fd424b8a0daab/send_stream.go#L211)?

Definitely not in the send stream, but yes, in the connection. That way, the kernel would handle the size of the read buffer.

I'm wondering how different packet sizes are dealt with when using GRO: For GRO, we'll pass a 64k buffer to the kernel. Does this mean that the kernel could (in principle) return one jumbo packet, or is there any way to set a maximum packet size?